e2e/assets/gitrepo/post-receive.sh

```diff
@@ -11,4 +11,12 @@ after=$(cat $ref)
 # necessary to make rancher/gitjob interpret the call as a push event coming from Github
 github_header="X-GitHub-Event: push"
 
-curl gitjob.cattle-fleet-system.svc.cluster.local -H "$github_header" -d "{\"ref\": \"$ref\", \"after\": \"$after\", \"repository\": {\"html_url\": \"{{.RepoURL}}\"}}"
+while ! curl --fail-with-body gitjob.cattle-fleet-system.svc.cluster.local \
+    -H "$github_header" \
+    -d "{\"ref\": \"$ref\", \"after\": \"$after\", \"repository\": {\"html_url\": \"{{.RepoURL}}\"}}"
+do
+    echo "Sending webhook failed, retrying..."
+    sleep 1
+done
```

Comment:
Depending on the `curl` version available, we could use some of the retry flags supported:
 - https://curl.se/docs/manpage.html#--retry
 - https://everything.curl.dev/usingcurl/downloads/retry

Reply:
I like that!